fitlins/interfaces/afni.py

```diff
@@ -0,0 +1,624 @@
+        # if nistats_flm.minimize_memory:
+        #     for key in results:
+        #         results[key] = SimpleRegressionResults(results[key])
+        # nistats_flm.results_.append(results)
+
+        return runtime
+
+    def save_remlfit_results(self, maps, contrasts, runtime):
+        """Parse  the AFNI "bucket" datasets written by 3dREMLfit and
+        subsequently read using nibabel. Save the results to disk according to
+        fitlins expectation.
+        Parameters
+        ----------
+        maps : A dictionary of nibabel.interfaces.brikhead.AFNIImage objects
+            keyed by output map type Description contrasts : TYPE Description
+            runtime : TYPE Description
+        contrasts : Object returned by nistats.contrasts.prepare_constrasts
+        runtime : nipype runtime object
+        """
+        import nibabel as nb
+
+        contrast_metadata = []
+        effect_maps = []
+        variance_maps = []
+        stat_maps = []
+        zscore_maps = []
+        pvalue_maps = []
+        fname_fmt = op.join(runtime.cwd, "{}_{}.nii.gz").format
+
+        out_ents = self.inputs.contrast_info[0]["entities"]
+
+        stats_img_info = parse_afni_ext(maps["stat"])
+
+        stat_types = np.array(
+            [
+                x.split("(")[0].replace("none", "").replace("test", "")
+                for x in stats_img_info["BRICK_STATSYM"].split(";")
+            ]
+        )
+
+        vol_labels = parse_afni_ext(maps["effect_size"])["BRICK_LABS"].split("~")
+
+        effect_bool = np.array([x.endswith("Coef") for x in vol_labels])
+        for (name, weights, contrast_type) in contrasts:
+            contrast_metadata.append(
+                {"contrast": name, "stat": contrast_type, **out_ents}
+            )
+
+            # Get boolean to index appropriate values
+            stat_bool = stat_types == contrast_type.upper()
+            contrast_bool = np.array([((name in x) and (('stim_' + name ) not in x)) for x in vol_labels])
+
+            # Get index for stdev
+            variance_img_labels = parse_afni_ext(maps["effect_variance"])[
+                "BRICK_LABS"
+            ].split("~")
+            stdev_bool = np.array(["StDev" == x for x in variance_img_labels])
+
+            # Indices for multi image nibabel object  should have length 1 and be integers
+            stat_idx = np.where(contrast_bool & stat_bool)[0]
+            stdev_idx = np.where(stdev_bool)[0]
+            # For multirow ftests there will be more than one index
+            effect_idx = np.where(contrast_bool & effect_bool)[0]
+
+            # Append maps:
+            # for each index into the result objects stored in maps, apply a
+            # modifying function as required and then append it to the
+            # appropriate output list type represented by map_list and write
+            # map_list to disk
+            for map_type, map_list, idx_list, mod_func in (
```

Comment:
mod_func doesn't appear to be used?

Reply:
hmmm... that would be an error and mean that the effect variance is currently incorrectly the stdev. I'll check this.